dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpBaseMessageImpl.java

```diff
@@ -2921,12 +2921,32 @@ else if (getServiceContext().getHttpConfig().onlySameSiteStar()) {
                         cookie.setSecure(true);
                     }
 
+                    // Set Partitioned Flag for SameSite=None Cookie
+                    if (getServiceContext().getHttpConfig().getPartitioned() == true
+                        && sameSiteAttributeValue.equalsIgnoreCase(HttpConfigConstants.SameSite.NONE.getName())) {
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc, "[1] Setting the Partitioned attribute for SameSite=None");
+                        }
+                        cookie.setAttribute("partitioned", "");
+                    }
+
                 } else {
                     if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                         Tr.debug(tc, "No SameSite configuration found");
                     }
                 }
             }
+            
+            // If SameSite=None is set programmatically, but partitioned is set via server.xml, then add the parititioned attribute
+            if (getServiceContext().getHttpConfig().useSameSiteConfig() && cookie.getAttribute("samesite") != null) {
+                boolean sameSiteNoneUsed = cookie.getAttribute("samesite").equalsIgnoreCase(HttpConfigConstants.SameSite.NONE.getName());
+                if(getServiceContext().getHttpConfig().getPartitioned() && sameSiteNoneUsed) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "[2] Setting the Partitioned attribute for SameSite=None");
+                    }
+                    cookie.setAttribute("partitioned", "");
```

Comment:
Just to verify this is the scenario where SameSite would have been set with setAttribute in the Servlet API or using set/addHeader in the Servlet API and defining a set-cookie header with samesite?

Reply:
this was added to for you scenario you gave:  `Do we need a test where we just set "partitioned=true" in the server.xml and then use the Servlet APIs to add samesite?`

Reply:
The first check (line 1925) looks if partitioned and samesite=none are set on the config. 

This second check (2941)  only looks if the samesite config is used and **the cookie** has samesite=none.   If so, we then check for partitioned and then add it if needed.